src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/CranBomTool.groovy

```diff
@@ -0,0 +1,135 @@
+ * under the License.
+ */
+package com.blackducksoftware.integration.hub.detect.bomtool
+
+import java.nio.charset.StandardCharsets
+
+import org.apache.commons.io.IOUtils
+import org.springframework.stereotype.Component
+
+import com.blackducksoftware.integration.hub.bdio.simple.model.DependencyNode
+import com.blackducksoftware.integration.hub.detect.bomtool.cran.PackratPackager
+import com.blackducksoftware.integration.hub.detect.type.BomToolType
+
+@Component
+class CranBomTool extends BomTool {
+    List<String> matchingSourcePaths = []
+    static final String PACKRAT_PATTERN = "packrat.lock"
+    static final String DESCRIPTION_PATTERN = "DESCRIPTION"
+
+    BomToolType getBomToolType() {
+        return BomToolType.CRAN
+    }
+
+    boolean isBomToolApplicable() {
+        //File sourceDirectory = new File(sourcePath)
+        if (sourcePathSearcher.findFilenamePattern(2, 'packrat.lock')){
+            matchingSourcePaths.addAll(sourcePathSearcher.findFilenamePattern(3, 'packrat.lock'))
+        }
+        else{
+            matchingSourcePaths.add("DNE")
+        }
+        if (sourcePathSearcher.findFilenamePattern(1, 'DESCRIPTION')){
+            matchingSourcePaths.addAll(sourcePathSearcher.findFilenamePattern(1, 'DESCRIPTION'))
+        }
+        else{
+            matchingSourcePaths.add("DNE")
+        }
+
+        !matchingSourcePaths.isEmpty()
+    }
+
+    List<DependencyNode> extractDependencyNodes() {
+        List<DependencyNode> projectNodes = []
+        int counter = 0
+        String sourcePath
+        File sourceDirectory
+        File[] packratLockFile
+        if (matchingSourcePaths[0] != "DNE"){
+            sourcePath = matchingSourcePaths[0]
+            sourceDirectory = new File(sourcePath)
+            packratLockFile = detectFileManager.findFilesToDepth(sourceDirectory, 'packrat.lock', 2)
+            counter ++
+        }
+        File[] descriptionFile
+        if(matchingSourcePaths[1] != "DNE"){
+            sourcePath = matchingSourcePaths[1]
+            sourceDirectory = new File(sourcePath)
+            descriptionFile = detectFileManager.findFilesToDepth(sourceDirectory, 'DESCRIPTION', 1)
+            counter --
+        }
+
+        switch (counter){
+            case 0:
+
+                final InputStream packratLockStream
+                final InputStream descriptionStream
+                try {
+                    packratLockStream = new FileInputStream(packratLockFile[0])
+                    descriptionStream = new FileInputStream(descriptionFile[0])
+                    String potentialProjectName = sourceDirectory.getName()
+                    String packratLock = IOUtils.toString(packratLockStream, StandardCharsets.UTF_8)
+                    String description = IOUtils.toString(descriptionStream, StandardCharsets.UTF_8)
+                    def packratPackager = new PackratPackager(projectInfoGatherer, nameVersionNodeTransformer)
```

Comment:
with groovy you could just called the "text" method on the packratLockFile and descriptionFile to get the text of a file as a string.